dialogue-core/src/test/java/com/palantir/dialogue/core/QueuedChannelTest.java

```diff
@@ -114,6 +115,26 @@ public void testQueuedRequestExecutedOnNextSubmission() {
         verify(delegate, times(3)).maybeExecute(endpoint, request);
     }
 
+    @Test
+    public void testQueuedRequestExecutedOnNextSubmission_throws() throws ExecutionException, InterruptedException {
+        Request queuedRequest = Mockito.mock(Request.class);
+        when(delegate.maybeExecute(endpoint, queuedRequest)).thenReturn(Optional.empty());
+        ListenableFuture<Response> queuedFuture = queuedChannel.execute(endpoint, queuedRequest);
+        verify(delegate, times(2)).maybeExecute(endpoint, queuedRequest);
+        assertThat(queuedFuture).isNotDone();
+
+        futureResponse.set(mockResponse);
```

Comment:
I would find it helpful to leave little comments explaining what you expect to be happening since just reading the test case requires tracing through the queuedChannel impl to understand why this is testing what it says it is

Reply:
good call, updated.